File: Remove the balls[ 14 - April - 2023]/Solution.java
//     >> JAVA CODE <<


class Solution {
    static class Pair{
        int color;
        int radius;
        
        Pair(int color,int radius)
        {
            this.color=color;
            this.radius=radius;
        }
    }
    public static int finLength(int N, int[] color, int[] radius) {
       
        Stack<Pair> stack=new Stack<>();
        
        for(int i=0;i<N;i++)
        {
            if(!stack.isEmpty())
            {
                if(stack.peek().color==color[i] && stack.peek().radius==radius[i])
                    stack.pop();
                else
                   stack.add(new Pair(color[i],radius[i]));
            }
            else
            {
                stack.add(new Pair(color[i],radius[i]));
            }
        }
        
        return stack.size();
    }
}

File: Remove the balls[ 14 - April - 2023]/Solution.cpp
//   >> C++ CODE <<


class Solution {
  public:
    
    class Pair {
        public:
            int color;
            int radius;
        
            Pair(int color, int radius) {
                this->color = color;
                this->radius = radius;
            }
    };
    
    int finLength(int N, vector<int>& color, vector<int>& radius) {
        stack<Pair> stack;
    
        for (int i = 0; i < N; i++) {
            if (!stack.empty()) {
                if (stack.top().color == color[i] && stack.top().radius == radius[i]) {
                    stack.pop();
                }
                else {
                    stack.push(Pair(color[i], radius[i]));
                }
            }
            else {
                stack.push(Pair(color[i], radius[i]));
            }
        }
    
        return stack.size();
}
};